### ISSUE TITLE
error: no viable conversion from 'script::Module' to 'std::shared_ptr<torch::jit::script::Module>'

### ISSUE THREAD

**Issue by aselva-eb** [pytorch/pytorch#23450]
Hi all,

I'm following the tutorial located https://pytorch.org/tutorials/advanced/cpp_export.html on Converting a PyTorch model to C++

I'm going through the minimal example as is on the page, but having a few issues when I try to make the binary at the end of step 3.

Specifically, here is the output error that I get:

`/Users/XXXXX/Desktop/cpp_transform/test/example-app.cpp:13:47: error: no viable conversion from  'script::Module' to 'std::shared_ptr<torch::jit::script::Module>'
std::shared_ptr<torch::jit::script::Module> module = torch::jit::load(argv[1]);`

among other warnings, and results in:

`1 error generated.
make[2]: *** [CMakeFiles/example-app.dir/example-app.cpp.o] Error 1
make[1]: *** [CMakeFiles/example-app.dir/all] Error 2
make: *** [all] Error 2`

How can I resolve this??

**Comment by driazati** [comment]
Are you using a nightly/master build of libtorch? This was changed recently and we haven't updated the tutorial on the stable docs yet (which apply to v1.1), you can see an updated version [here](https://github.com/pytorch/tutorials/pull/560).

TLDR: `script::Module` was changed to a value type, so you can just do `script::Module myModule = torch::jit::load(argv[1]);`

Closing since we already have a PR out to fix the tutorial, but feel free to re-open if you still have problems

**Comment by aselva-eb** [comment]
Thanks, I updated my code according to the changes in the PR you linked to.

I was able to get a bit farther, but now I'm running into other problems with the libtorch library. It looks like the dependencies weren't built for the arch of my Mac OSX: 
When I go to make my application...

`Undefined symbols for architecture x86_64:
  "c10::Error::Error(c10::SourceLocation, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      c10::intrusive_ptr<c10::ivalue::Object, c10::detail::intrusive_target_default_null_type<c10::ivalue::Object> >::retain_() in example-app.cpp.o`

I tried the latest stable release of libtorch (specifically selected Mac OSX), but it doesn't have any of the shared dependencies that I need to get it working. The version of libtorch I downloaded was from here: https://pytorch.org/cppdocs/installing.html , it looks like it gets my farther.. but then I get stuck with dependency issues for OSX.

**Comment by aselva-eb** [comment]
I resolved this issue by building PyTorch from source, and using the location of that lib when I use cmake to generate the make files.

i.e `cmake -DCMAKE_PREFIX_PATH=/Users/XXXXXX/Desktop/cpp_transform/libtorch -DCMAKE_C_COMPILER="/usr/local/opt/llvm/bin/clang" -DCMAKE_CXX_COMPILER="/usr/local/opt/llvm/bin/clang++" ..`

Also, since OSX clang didn't support fopenmp which was required for building, I had to `brew install llvm` and use its clang binary instead of the native one (hence the override for `DCMAKE_C_COMPILER` and `DCMAKE_CXX_COMPILER`

Hope this info might be useful for anyone else going through the minimal example on PyTorch's website and are building for OSX. Would be great if the instructions were revised to include these considerations!

Thanks!
